### PR TITLE
Remove Group Parameters

### PR DESCRIPTION
**Pull request by EwenQuim**
This PR aims to remove Group Parameters in favor of reusable Route Options. While it seems to be a regression, it is more in the spririt of Go:

- Group parameters are a kind of heritance feature
- Route options are more a "composable" approach
- Simpler code to maintain. In the next release, I will focus on making Fuego's code more simple & elegant

@dylanhitt could you please review it?

### CODE REVIEW

File: serialization_test.go [hunk 1]
```diff
@@ -64,6 +65,8 @@ func TestRecursiveJSON(t *testing.T) {
 		err := SendJSON(w, nil, value)
 
 		require.Error(t, err)
+		var unsupportedValueError *json.UnsupportedValueError
```
Comment (dylanhitt) [inline serialization_test.go:68]: Is this left over from #200? Not sure how it relates to this PR

Comment (EwenQuim) [reply serialization_test.go:68]: Yes sorry

**Comment by EwenQuim** [comment]
> If the plan is to eventually add the options style configuration on to the Server/Group. I'd be 100% on board.

Yes I hesitated, we might let the final decision to the users.

We might do this and **keep the same signature** as the route options, which will not be trivial.

**Comment by EwenQuim** [comment]
Closed in favor of #233

**Pull request closed without merging**